pkg/ddl/backfilling.go

```diff
@@ -619,6 +621,146 @@ func SetBackfillTaskChanSizeForTest(n int) {
 	backfillTaskChanSize = n
 }
 
+func (dc *ddlCtx) runAddIndexInLocalIngestMode(
+	ctx context.Context,
+	sessPool *sess.Pool,
+	t table.PhysicalTable,
+	reorgInfo *reorgInfo,
+) error {
+	// TODO(tangenta): support adjust worker count dynamically.
+	startKey, endKey := reorgInfo.StartKey, reorgInfo.EndKey
+	if err := dc.isReorgRunnable(reorgInfo.Job.ID, false); err != nil {
+		return errors.Trace(err)
+	}
+	job := reorgInfo.Job.Clone()
+	ctx = tidblogutil.WithCategory(ctx, "ddl-ingest")
+	opCtx := NewLocalOperatorCtx(ctx, job.ID)
+	bcCtx, err := getBackendCtx(ctx, dc.store, dc.etcdCli, job)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	idxCnt := len(reorgInfo.elements)
+	indexIDs := make([]int64, 0, idxCnt)
+	indexInfos := make([]*model.IndexInfo, 0, idxCnt)
+	uniques := make([]bool, 0, idxCnt)
+	for _, e := range reorgInfo.elements {
+		indexIDs = append(indexIDs, e.ID)
+		indexInfo := model.FindIndexInfoByID(t.Meta().Indices, e.ID)
+		if indexInfo == nil {
+			logutil.DDLIngestLogger().Warn("index info not found",
```

Comment:
not sure why we have many types of logger (line 636, 661)

Reply:
Unified with `tidblogutil.Logger(ctx)`

Reply:
Maybe `logutil.DDLIngestLogger()` is better :)

https://github.com/pingcap/tidb/pull/54149/commits/b81b2c3e10ce84a122ec6d2f4e35581452b840d2